setup.sh

```diff
@@ -49,51 +111,130 @@ cat <<EOF
   }
 }
 EOF
-echo ""
+  )
 
+  consensus="./tezos_interop/consensus.mligo"
+  storage=$(ligo compile storage "$consensus" "$storage")
+  contract=$(ligo compile contract $consensus)
 
-validators_json () {
-  ## most of the noise in this function is because of indentation
-  echo "["
-for VALIDATOR in "${VALIDATORS[@]}"; do 
-  i=$(echo $VALIDATOR | awk -F';' '{ print $1 }')
-  KEY=$(echo $VALIDATOR | awk -F';' '{ print $2 }')
-  URI=$(echo $VALIDATOR | awk -F';' '{ print $3 }')
-  if [ $i != 0 ]; then
-    printf ",
-"
+  $node_manager client originate contract "consensus" \
+    transferring 0 from myWallet \
+    running "$contract" \
+    --init "$storage" \
+    --burn-cap 2 \
+    --force
+
+  for VALIDATOR in ${VALIDATORS[@]}; do
+    i=$(echo $VALIDATOR | awk -F';' '{ print $1 }')
+    FOLDER="$data_directory/$i"
+
+    validators_json > "$FOLDER/validators.json"
+    trusted_validator_membership_change_json > "$FOLDER/trusted-validator-membership-change.json"
+  done
+
+  TEZOS_CONSENSUS_ADDRESS="$($node_manager client show known contract consensus | grep KT1 | tr -d '\r')"
+  TEZOS_RPC_NODE="http://localhost:8732"
+  TEZOS_SECRET="$($node_manager client show address myWallet -S | tail -n 1 | cut -c 25- | tr -d '\r')"
+
+  message "Configuring Deku nodes"
+  for VALIDATOR in ${VALIDATORS[@]}; do
+    i=$(echo $VALIDATOR | awk -F';' '{ print $1 }')
+    FOLDER="$data_directory/$i"
+
+    sidecli setup-tezos "$FOLDER" \
+      --tezos_consensus_contract="$TEZOS_CONSENSUS_ADDRESS" \
+      --tezos_rpc_node="$TEZOS_RPC_NODE" \
+      --tezos_secret="$TEZOS_SECRET"
+  done
+  message "Done! Go beyond!"
+}
+
+start_node() {
+  # Get a new snapshot if no snapshot exists or current is older than 1 day
+  # Snapshots drastically reduce the bootstrapping time of a node by allowing
+  # it to import an existing state instead of downloading every block from the
+  # network.
+  if [[ ! -e "./snapshot" || $(find "./snapshot" -mtime +1 -print) ]]; then
+    message "Fetching Snapshot "
+    curl -L https://granadanet.xtz-shots.io/rolling -o ./snapshot
+    message "Importing Snapshot "
+    $node_manager snapshot import "$PWD/snapshot"
   fi
-cat <<EOF
-  {
-    "address": "$KEY",
-    "uri": "$URI"
-EOF
-  printf "  }"
-done
+
+  message "Starting Node "
+  $node_manager stop
+  $node_manager node start --rpc-port 8732 --allow-all-rpc :8732
+
+  message "Waiting for Bootstrap "
+  sleep 5
+  $node_manager client bootstrapped >/dev/null
```

Comment:
I wonder if people may find it useful to see the output so they can see the node is bootstrapping and not just hanging?
```suggestion
  $node_manager client bootstrapped
```